### PR TITLE
Api refactorization

### PR DESCRIPTION
**Pull request by mbredif**
This PR refactors the previous api methods (one per li3ds object type) to publish dictionaries into api objects (one per li3ds object type as well), to decorrelate object creation (the definition of its parameters) from publication through the api.

### CODE REVIEW

File: micmac_li3ds/import_autocal.py [hunk 1]
```diff
@@ -2,9 +2,8 @@
 import logging
 
 from cliff.command import Command
-from argparse import Namespace
 
-from . import api as li3ds
+from . import api as Api
```
Comment (elemoine) [inline micmac_li3ds/import_autocal.py]: By convention module names start with a lower-case character, while class names start with an upper-case character. So please use something else than `Api` here for the alias to the `api` namespace. You probably don't need an alias anyway. You could use `api` to refer to the module name, and `apiobj` for the `api.Api` instance.

File: micmac_li3ds/import_autocal.py [hunk 2]
```diff
@@ -56,177 +46,186 @@ def get_parser(self, prog_name):
...
 
-    def take_action(self, args):
+    def take_action(self, parsed_args):
         """
         Create or update a camera sensor.
         """
-        api = li3ds.Api(args.api_url, args.api_key, args.no_proxy,
-                        self.log, args.indent)
-        for filename in args.filenames:
+        api = Api.Api(parsed_args, self.log)
```
Comment (elemoine) [inline micmac_li3ds/import_autocal.py]: So it should be `apiobj = api.Api(parsed_args, self.log)`.

File: micmac_li3ds/import_autocal.py [hunk 3]
```diff
@@ -56,177 +46,186 @@ def get_parser(self, prog_name):
...
-    return api.get_or_create_transfotree(
-        name=args.transfotree,
-        transfos=transfos,
-        owner=args.owner,
-    )
+            ApiObjs(args, filename).get_or_create(api)
+            self.log.info('Success!\n')
+
+
+class ApiObjs(Api.ApiObjs):
```
Comment (elemoine) [inline micmac_li3ds/import_autocal.py]: Should be `class ApiObjs(api.ApiObjs):`.

File: micmac_li3ds/import_autocal.py [hunk 4]
```diff
@@ -56,177 +46,186 @@ def get_parser(self, prog_name):
...
+    def __init__(self, args, filename, node=None):
+        if node:
+            print(node.tag)
```
Comment (elemoine) [inline micmac_li3ds/import_autocal.py]: `print` to be removed, or replaced with an appropriate log statements.

File: micmac_li3ds/import_autocal.py [hunk 5]
```diff
@@ -56,177 +46,186 @@ def get_parser(self, prog_name):
...
+                      '+Z: optical axis (in front of the camera), ' \
+                      '{description}'
+        return Api.Referential(
+            sensor, referential,
+            name='camera',
+            description=description.format(**referential),
+        )
+
+
+class Transfo:
```
Comment (elemoine) [inline micmac_li3ds/import_autocal.py]: You don't need a class here. Why not just defining `create_pinhole`, `create_orintglob_transfo` and `create_distortion_transfo` functions instead?

File: micmac_li3ds/import_orimatis.py [hunk 6]
```diff
@@ -156,261 +207,159 @@ def get_acquisition_datetime(node):
...
+            },
         )
 
-    def get_or_create_wo_referential(self, node, sensor):
-
-        node = xmlutil.child(node, 'geometry/extrinseque')
-        systeme = xmlutil.child(node, 'systeme').text.strip()
-        grid_alti = xmlutil.child(node, 'grid_alti').text.strip()
 
+class Referential:
```
Comment (elemoine) [inline micmac_li3ds/import_orimatis.py]: Please do not use classes as namespaces. I think you don't need a namespace here. You can just add functions to the `import_orimatis` module.

File: micmac_li3ds/import_orimatis.py [hunk 7]
```diff
@@ -156,261 +207,159 @@ def get_acquisition_datetime(node):
...
         )
 
-    def get_or_create_pinh_transform(self, node, source, target):
-        node = xmlutil.child(node, 'geometry/intrinseque/sensor')
-        name = '{}#intrinseque'.format(self.transfotree)
-        return self.api.get_or_create_transfo(
-            name, 'projective_pinhole', source, target,
-            description='imported from "{}"'.format(self.basename),
+
+class Transfo:
```
Comment (elemoine) [inline micmac_li3ds/import_orimatis.py]: `Transfo` class to be removed.

**Comment by elemoine** [comment]
Also, I'd appreciate if you could clean up the commit history.

File: micmac_li3ds/import_autocal.py (hunk 1 above)
Comment (mbredif) [inline micmac_li3ds/import_autocal.py]: done

File: micmac_li3ds/import_autocal.py (hunk 2 above)
Comment (mbredif) [inline micmac_li3ds/import_autocal.py]: done, using `server = api.ApiServer(...)`

File: micmac_li3ds/import_autocal.py (hunk 3 above)
Comment (mbredif) [inline micmac_li3ds/import_autocal.py]: done

File: micmac_li3ds/import_autocal.py (hunk 5 above)
Comment (mbredif) [inline micmac_li3ds/import_autocal.py]: Done, I switched all such cases with `{type}_{name}` functions, such as `transfo_pinhole(...)`

**Comment by elemoine** [comment]
You've added `import_rosbag.py` by mistake I guess.

File: micmac_li3ds/import_autocal.py (hunk 4 above)
Comment (mbredif) [inline micmac_li3ds/import_autocal.py]: done

File: micmac_li3ds/import_autocal.py (hunk 5 above)
Comment (mbredif) [inline micmac_li3ds/import_autocal.py]: done

File: micmac_li3ds/import_orimatis.py (hunk 6 above)
Comment (mbredif) [inline micmac_li3ds/import_orimatis.py]: done

File: micmac_li3ds/import_orimatis.py (hunk 7 above)
Comment (mbredif) [inline micmac_li3ds/import_orimatis.py]: done

**Comment by mbredif** [comment]
I tried to take your comments into account and slightly cleaned up the history. Feel free to merge and work on a PR if you have further concerns.

**Comment by mbredif** [comment]
Good catch, I removed the WIP `import_rosbag.py`. Tests are ok now.

**Pull request merged**